tests/integration/utils.go

```diff
@@ -593,19 +592,13 @@ func areMapsSame(mp1, mp2 map[string]string) bool {
 	return true
 }
 
-func compareBuckets(list1 []types.Bucket, list2 []s3response.ListAllMyBucketsEntry) bool {
+func compareBuckets(list1 []types.Bucket, list2 []types.Bucket) bool {
 	if len(list1) != len(list2) {
 		return false
 	}
 
-	elementMap := make(map[string]bool)
-
-	for _, elem := range list1 {
-		elementMap[*elem.Name] = true
-	}
-
-	for _, elem := range list2 {
-		if _, found := elementMap[elem.Name]; !found {
+	for i, elem := range list1 {
+		if *elem.Name != *list2[i].Name {
```

Comment:
This requires lists to both be sorted the same. Is that an ok assumption here?

Reply:
Yes, exactly.

The comparison requires both lists to be sorted, as ListBuckets returns a sorted result. Based on this assumption, we also verify the order of the bucket names.